chain/network/tests/routing.rs

```diff
@@ -599,6 +619,61 @@ fn ping_jump() {
     .unwrap();
 }
 
+/// Test routed messages are not dropped if have enough TTL.
+/// Spawn three nodes and connect them in a line:
+///
+/// 0 ---- 1 ----- 2
+///
+/// Set routed message ttl to 2, so routed message can't pass through more than 2 edges.
+/// Send Ping from 0 to 2. It should arrive since there are only 2 edges from 0 to 2.
+/// Check Ping arrive at node 2 and later Pong arrive at node 0.
+#[test]
+fn test_dont_drop_after_ttl() {
+    init_test_logger();
+
+    System::run(|| {
+        let mut runner = Runner::new(3, 1, 3).routed_message_ttl(2);
+
+        runner.push(Action::AddEdge(0, 1));
+        runner.push(Action::AddEdge(1, 2));
+        runner.push(Action::CheckRoutingTable(0, vec![(1, vec![1]), (2, vec![1])]));
+        runner.push(Action::PingTo(0, 0, 2));
+        runner.push(Action::CheckPingPong(2, vec![(0, 0)], vec![]));
+        runner.push(Action::CheckPingPong(0, vec![], vec![(0, 2)]));
+
+        runner.run();
+    })
+    .unwrap();
+}
+
+/// Test routed messages are dropped if don't have enough TTL.
+/// Spawn three nodes and connect them in a line:
+///
+/// 0 ---- 1 ----- 2
+///
+/// Set routed message ttl to 1, so routed message can't pass through more than 1 edges.
+/// Send Ping from 0 to 2. It should not arrive since there are 2 edges from 0 to 2.
+/// Check none of Ping and Pong arrived.
+#[test]
+fn test_drop_after_ttl() {
+    init_test_logger();
+
+    System::run(|| {
+        let mut runner = Runner::new(3, 1, 3).routed_message_ttl(1);
+
+        runner.push(Action::AddEdge(0, 1));
+        runner.push(Action::AddEdge(1, 2));
+        runner.push(Action::CheckRoutingTable(0, vec![(1, vec![1]), (2, vec![1])]));
+        runner.push(Action::PingTo(0, 0, 2));
+        runner.push(Action::Wait(100));
+        runner.push(Action::CheckPingPong(2, vec![], vec![]));
+        runner.push(Action::CheckPingPong(0, vec![], vec![]));
+
+        runner.run();
```

Comment:
How exactly does it check that the message didn't arrive?

Reply:
I already fixed what CheckPingPong does to match exactly what is passed in arguments. In this case, 0 ping/pongs received.

In this test I set ttl to 1, message will not go over two edges and then send a ping from 0 to 2.
Wait for 100ms. This is more than enough in this test for message to arrive from 0 to 2, but it never arrives since message is dropped by 1. Last to actions are checking that ping/pong didn't get to destination (in this case ping would be enough, since pong action won't be even triggered).

Notice that test above this is the same but with ttl equals 2 and message does arrive at their destination.